compiler/rustc_public/src/lib.rs

```diff
@@ -41,7 +41,7 @@ use crate::unstable::Stable;
 
 pub mod abi;
 mod alloc;
-pub(crate) mod unstable;
+pub mod unstable;
```

Comment:
I don't think we should make this module public. Instead, we should add an attribute to the trait declarations:
```rust
/// This is used by the public `rustc_internal::stable` function when feature `rustc_internal` is enabled.
#[cfg_attr(not(feature = "rustc_internal"), allow(unreachable_pub))]
pub trait Stable<'tcx>: PointeeSized {
  // ...
}
```